backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/trace/repository/RetrieveDbRecordsUtil.java

```diff
@@ -503,6 +502,17 @@ static IQueryBuilder<I_M_HU_Trace> createQueryBuilderOrNull(@NonNull final HUTra
 				queryIsEmpty = false;
 			}
 
+			if (query.getIsActive() != null)
+			{
+				queryBuilder.addEqualsFilter(I_M_HU_Trace.COLUMNNAME_IsActive, query.getIsActive());
+				queryIsEmpty = false;
+			}
+			else
+			{
+				// dev-note: for all the other cases where query is not empty (queryIsEmpty = false), consider only the active records
+				queryBuilder.addOnlyActiveRecordsFilter();
```

Comment:
but in case query.getIsActive() is null
then we do exactly what we would do if query.getIsActive() is true.

so I am wondering what the actual reason we need HUTraceEventQuery.isActive to be nullable `Boolean`?
why not `@Default boolean isActive=true;` ?

Reply:
clarified